### PR TITLE
CVSL-132: Combine nomis and delius to build a caseload for the currently logged in user

### PR DESCRIPTION
**Pull request by smcveigh941**
* Shifted all the type declarations to the `@types` folder
* Implemented the prisoner search endpoint to search the nomisIds returned in caseload from delius.
* Combined data from nomis and delius to create a single type `ManagedCase` with all of the information about a case
* We should think about caching this caseload for use later

### CODE REVIEW

File: server/data/prisonerSearchApiClient.ts [hunk 1]
```diff
@@ -10,9 +11,16 @@ export default class PrisonerSearchApiClient {
   }
 
   async searchPrisoners(prisonerSearchCriteria: PrisonerSearchCriteria): Promise<Prisoner[]> {
-    return this.restClient.post({
+    return (await this.restClient.post({
```
Comment (yasinmustafa) [inline server/data/prisonerSearchApiClient.ts:14]: I dont think you need these awaits at this level - if you await at the calling function that should suffice

**Review by mikehalmamoj — DISMISSED** [review]

**Review by yasinmustafa — DISMISSED** [review]
LGTM - one minor comment

File: integration_tests/mockApis/prison.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { SuperAgentRequest } from 'superagent'
+import { stubFor } from '../wiremock'
+
```
Comment (timharrison-moj) [inline integration_tests/mockApis/prison.ts:3]: As there are two APIs we'll use in prisons - maybe a name change to `prisonerSearchApi` for this one? Other one can be something like `prisonApi`? Just clarifying which is which?

File: server/routes/creatingLicences/handlers/caseload.test.ts [hunk 3]
```diff
@@ -47,9 +44,9 @@ describe('Route Handlers - Create Licence - Caseload', () => {
       expect(res.render).toHaveBeenCalledWith('pages/create/caseload', {
         caseload: [
           {
-            name: 'Jones',
+            name: 'Joe Rogan',
             crnNumber: 'X381306',
-            conditionalReleaseDate: '03 August 2022',
+            conditionalReleaseDate: '09/09/2022',
           },
```
Comment (timharrison-moj) [inline server/routes/creatingLicences/handlers/caseload.test.ts:50]: Might be some content challenge on date formats - but not sure we're had a steer on consistent date format yet!

File: server/services/caseloadService.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    return (await this.prisonerService.searchPrisonersByNomisIds(username, caseloadNomisIds)).map(offender => {
+      const matchingDeliusCase = managedOffenders.find(deliusCase => deliusCase.nomsNumber === offender.prisonerNumber)
+      return {
+        ...matchingDeliusCase,
+        ...offender,
+      } as ManagedCase
+
+      // TODO Cache this by username?
+    })
```
Comment (timharrison-moj) [inline server/services/caseloadService.ts:34]: Can cache in redis later and wrap the get with a redis lookup first - good examples in other services to reuse.

**Review by timharrison-moj — APPROVED** [review]
LGTM - tiny comment about naming only.

**Review by timharrison-moj — APPROVED** [review]
LGTM.

**Pull request merged**